lib/nitcorn/logging.nit

```diff
@@ -0,0 +1,58 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# logs requests
+module logging
+
+import reactor
+import realtime
+import performance_analysis
+
+redef class Action
+
+	redef fun prepare_respond_and_close(request, truncated_uri, http_server) do
+		if debug then
+			print """{{{class_name}}} enter:
+uri="{{{truncated_uri}}}"
+query="{{{request.query_string}}}"
+body:{{{request.body.length}}} bytes"""
+			var clock = new Clock
+			super
+			var perf = sys.perfs[class_name]
+			perf.add(clock.lapse)
+			if perf.count % perfs_print_period == 0 then print "{class_name} perfs: {perf}:"
+			print "{class_name} return: uri={truncated_uri}"
+		else
+			super
+		end
+	end
+end
+
+redef fun print(object) do
+	var timestamp = new Tm.gmtime
+	super "{timestamp.year}/{timestamp.mon}/{timestamp.mday} "+
+	"{timestamp.hour}:{timestamp.min}:{timestamp.sec}: {object}"
```

Comment:
It is already in a separate module, it is not imported from `nitcorn::nitcorn`. In my opinion, it should be imported by default, because you need it before knowing that you need it. But I'm OK to keep it optional until we have a change to test it out. 

`ts` is a nice tool but it requires more work to use. I'm not even sure how to apply it to stderr without mergin stdout and stderr.